{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "compute-a-17-122.o2.rc.hms.harvard.edu\n",
      "HSM Computational partition detected.\n",
      "/n/groups/reich/hringbauer/git/y_chrom\n",
      "CPU Count: 32\n",
      "3.7.4 (default, Sep 11 2019, 11:24:51) \n",
      "[GCC 6.2.0]\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "import os  # For Saving to Folder\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "import socket\n",
    "import os as os\n",
    "import sys as sys\n",
    "import multiprocessing as mp\n",
    "\n",
    "### For Arial Font\n",
    "from matplotlib import rcParams\n",
    "rcParams['font.family'] = 'sans-serif'   # Set the defaul\n",
    "### Make sure to have the font installed (it is on cluster for Harald)\n",
    "rcParams['font.sans-serif'] = ['Arial']\n",
    "\n",
    "socket_name = socket.gethostname()\n",
    "print(socket_name)\n",
    "\n",
    "if socket_name.startswith(\"compute-\"):\n",
    "    print(\"HSM Computational partition detected.\")\n",
    "    path = \"/n/groups/reich/hringbauer/git/y_chrom/\"  # The Path on Midway Cluster\n",
    "else:\n",
    "    raise RuntimeWarning(\"Not compatible machine. Check!!\")\n",
    "\n",
    "os.chdir(path)  # Set the right Path (in line with Atom default)\n",
    "# Show the current working directory. Should be HAPSBURG/Notebooks/ParallelRuns\n",
    "print(os.getcwd())\n",
    "print(f\"CPU Count: {mp.cpu_count()}\")\n",
    "print(sys.version)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Define Parameters for param file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_parfile_from_ref(iid, \n",
    "                            savepath = \"/n/groups/reich/hringbauer/git/y_chrom/parfiles/\",\n",
    "                            outfolder = \"\",\n",
    "                            df=[],\n",
    "                            col_id = 'Version ID',\n",
    "                            bam_path = \"\",\n",
    "                            snp_file = \"\"):  #/n/groups/reich/hringbauer/git/y_chrom/data/eigenstrat/y_snps_all2020.snp\n",
    "    \"\"\"Create a new parfile from reference dataframe df\n",
    "    iid: What individual to prepare\n",
    "    id_col: What column to use in anno file for IID\n",
    "    savepath: Path to save the new parfile to\n",
    "    bam_path: Path of the BAM File to use. If not given, \n",
    "    find out from df.\n",
    "    outfolder: p\n",
    "    snp_file: What SNP file to use (which SNPs to pulldown)\"\"\"\n",
    "    par_dc = give_default_par_dc()  ### Get the default Parameter Dictionary\n",
    "    \n",
    "    if len(bam_path)==0:\n",
    "        bam_path = extram_bam_df(df, iid=iid, col_id=col_id)\n",
    "    \n",
    "    if len(snp_file)>0:\n",
    "        par_dc[\"snpname\"] =  snp_file\n",
    "        #par_dc[\"readdepthname\"] = '/n/groups/reich/hringbauer/git/y_chrom/output/pulldowns_all/SAMP.txt'\n",
    "    \n",
    "    if os.path.isdir(bam_path):\n",
    "        bam_path = os.path.join(bam_path, \"*Y.bam\")  # Append the Y BAM\n",
    "        \n",
    "    if os.path.isdir(savepath):\n",
    "        savepath = os.path.join(savepath, iid)\n",
    "    #else:\n",
    "        #bam_path = bam_path + \".by_chr/*Y.bam\"   # Hack for Ning samples. Not clear how general\n",
    "        #print(\"Warning. Ning2020 hack run. General?\")\n",
    "        \n",
    "    par_dc[\"SAMP\"] = iid\n",
    "    par_dc[\"bamname\"] = bam_path\n",
    "    if len(outfolder)>0:\n",
    "        par_dc[\"readdepthname\"] = os.path.join(outfolder ,\"SAMP.txt\")\n",
    "        \n",
    "    savepath = create_parfile(par_dc, savepath=savepath)\n",
    "    return savepath\n",
    "\n",
    "def create_parfile(par_dc, savepath=\"/n/groups/reich/hringbauer/git/y_chrom/parfiles/\"):\n",
    "    \"\"\"Creates parfile from dictonary of Parameters\n",
    "    savepath: Where to save the parifle to\"\"\"\n",
    "    ### Modify Parameter Dictionary\n",
    "    with open(savepath,\"w\") as f:\n",
    "        for k, v in par_dc.items():        \n",
    "            f.write(f\"{k}: {v}\\n\")\n",
    "    print(f\"Successfully saved parfile to {savepath}\")\n",
    "    return savepath\n",
    "    \n",
    "def run_pulldown_y(path_parfile, path_bin = \"./scripts/pulldown\", output=True):\n",
    "    \"\"\"Run Pulldown via par file.\n",
    "    path_parfile: Which parfile to run\n",
    "    path_bin: Where to find the binary to run it\"\"\"\n",
    "    if output:\n",
    "        ! $path_bin -p $path_parfile\n",
    "    else:\n",
    "        ! $path_bin -p $path_parfile  > /dev/null\n",
    "    \n",
    "def extram_bam_df(df, iid,\n",
    "                  col_id = 'Instance ID (\"_all\" means includes a mix of UDG-treated and non-UDG-treated; \"_published\" distinguishes a published sample for a still-unpublished higher quality version)', \n",
    "                  col_bam = 'Data: autosomal bam'):\n",
    "    \"\"\"Return bam Path from Meta File\"\"\"\n",
    "    idx = (df[col_id]==iid)\n",
    "    assert(np.sum(idx)>0) \n",
    "    bam_path = df.loc[idx, col_bam].values[0]\n",
    "    return bam_path\n",
    "\n",
    "def give_default_par_dc():\n",
    "    \"\"\"Return the default parameter dictionary\"\"\"\n",
    "    par_dc = {\"BASE\": \"/n/groups/reich/matt/pipeline/static\",\n",
    "          \"SAMP\": \"FILL_IN\",\n",
    "          \"D2\": \"./output\",\n",
    "          \"indivname\": \"/n/groups/reich/DAVID/V43/V43.5/v43.5.ind\",\n",
    "          \"snpname\": \"/n/groups/reich/matt/pipeline/static/1240kSNP.snp\",\n",
    "          \"bamname\": \"FILL_IN\",\n",
    "          \"threshtable\": \"/n/groups/reich/matt/pipeline/static/pulldown_thresholds\",       \n",
    "          \"defstring\": \"capture_half\",\n",
    "          \"readbam\": \"/n/groups/reich/matt/pipeline/static/readbam\",\n",
    "          \"oldpullmode\": \"YES\",\n",
    "          \"sampname\": \"SAMP\",\n",
    "          \"printcount\": \"NO\",\n",
    "          \"majmode\": \"NO\",\n",
    "          \"rgonly\": \"NO\",\n",
    "          \"chrom\": \"24\",\n",
    "          \"readdepthname\": \"/n/groups/reich/hringbauer/git/y_chrom/output/SAMP.txt\"\n",
    "          }\n",
    "    return par_dc"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Run for a single Individual"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%time\n",
    "#anno_path = \"/n/groups/reich/hringbauer/explore_ntbk/v43/V43/v42.3.anno\"\n",
    "anno_path = \"/n/groups/reich/DAVID/V43/V43.5/v43.5.anno\"\n",
    "df = pd.read_csv(anno_path, sep='\\t', low_memory=False)\n",
    "\n",
    "### Create a parfile to run\n",
    "savepath = create_parfile_from_ref(df=df, \n",
    "                                   iid=\"I10266\",\n",
    "                                   outfolder=\"./output/test/\")\n",
    "### Produce Lots of Output (toggled)\n",
    "run_pulldown_y(path_parfile=savepath)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/n/groups/reich/hringbauer/git/y_chrom/parfiles/I10266\n"
     ]
    }
   ],
   "source": [
    "print(savepath)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Do Multiple Individuals"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Doing SEC002\n",
      "/n/data1/hms/genetics/reich/1000Genomes/amh_samples/marcus2019_sardinia_samples/A-round1/B-fix/SEC002/hg19/SEC002.md_no1kgvar.bam.by_chr/*Y.bam\n",
      "Successfully saved to /n/groups/reich/hringbauer/git/y_chrom/parfiles/SEC002\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/marcus2019_sardinia_samples/A-round1/B-fix/SEC002/hg19/SEC002.md_no1kgvar.bam.by_chr/*Y.bam Y \n"
     ]
    }
   ],
   "source": [
    "#iids = [\"I7116\", \"I7121\", \"I7120\"]  # African R1b\n",
    "#iids = [\"R6.SG\", \"I18872\", \"I15941\",\"I8016\",\"I6752\", \"I16339\", \"I14337\", \"I8057\", \"I7116\"] #R1b-V88 canditates\n",
    "iids = [\"SEC002\"]\n",
    "\n",
    "\n",
    "for iid in iids:\n",
    "    print(f\"Doing {iid}\")\n",
    "    savepath = create_parfile_from_ref(df=df, iid=iid,\n",
    "                            par_dc=par_dc)\n",
    "    run_pulldown_y(path_parfile=savepath, output=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Run the Punic Individuals"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#punic_df = pd.read_csv(\"../punic_aDNA/data/males_feb20.csv\")\n",
    "punic_df = pd.read_csv(\"../punic_aDNA/data/males_sic_feb20.csv\")\n",
    "\n",
    "male_punic_iids = punic_df.iloc[:,0].values\n",
    "\n",
    "for iid in male_punic_iids:\n",
    "    print(f\"Doing {iid}\")\n",
    "    savepath = create_parfile_from_ref(df=df, iid=iid,\n",
    "                            par_dc=par_dc)\n",
    "    \n",
    "    run_pulldown_y(path_parfile=savepath, output=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Pull down Ning Individuals"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded 30 from /n/groups/reich/hringbauer/git/tibet_aDNA/Data/iids_male_ning2020.csv\n"
     ]
    }
   ],
   "source": [
    "iid_path = \"/n/groups/reich/hringbauer/git/tibet_aDNA/Data/iids_male_ning2020.csv\"\n",
    "iids = np.loadtxt(iid_path, dtype=\"str\")\n",
    "print(f\"Loaded {len(iids)} from {iid_path}\")\n",
    "\n",
    "anno_path = \"/n/groups/reich/DAVID/V43/V43.5/v43.5.anno\"\n",
    "df = pd.read_csv(anno_path, sep='\\t', low_memory=False)\n",
    "\n",
    "bam_folder = \"/n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Test One Indiviual"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "iid=\"XW-M1R18\"\n",
    "savepath = create_parfile_from_ref(iid=iid,  #I10266\n",
    "                                   bam_path= bam_folder + iid + \".Y.bam\",\n",
    "                                   outfolder=\"/n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/\")\n",
    "run_pulldown_y(path_parfile=savepath) # Run the Pulldown"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Pull down many individuals"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/XW-M1R18\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/XW-M1R18\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: XW-M1R18\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/XW-M1R18.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: XW-M1R18\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/XW-M1R18.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 1982602937\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: XW-M1R18\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/XW-M1R18.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "./data/eigenstrat/y_snps_all2020_vO.snp: genetic distance set from physical distance\n",
      "number of valid snps: 73680\n",
      "sample XW-M1R18 is male\n",
      "numsnps: 73680  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for XW-M1R18 (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0        XW-M1R18  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/XW-M1R18.Y.bam Y \n",
      "triallelic: 738\n",
      "rawreads: 543216\n",
      "reg: 24  reads: 543216\n",
      "chrom 24 processed\n",
      "            XW-M1R18          China_YR_MN mean depth:     2.322340 coverage: 61839\n",
      "histogram of read return codes\n",
      "read return code -166 ::      7640\n",
      "read return code   -5 ::    344905\n",
      "read return code   -2 ::     41903\n",
      "read return code   -1 ::      2550\n",
      "read return code    1 ::    146218\n",
      "total: 543216\n",
      "\n",
      "## total reads: 543216  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 73680\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 61838 invalids 0\n",
      "\n",
      "##end of pulldown:        6.490 seconds cpu       47.649 Mbytes in use\n",
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/WGM43\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/WGM43\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: WGM43\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/WGM43.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: WGM43\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/WGM43.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 166907343\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: WGM43\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/WGM43.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "./data/eigenstrat/y_snps_all2020_vO.snp: genetic distance set from physical distance\n",
      "number of valid snps: 73680\n",
      "sample WGM43 is male\n",
      "numsnps: 73680  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for WGM43 (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0           WGM43  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/WGM43.Y.bam Y \n",
      "triallelic: 15\n",
      "rawreads: 4570\n",
      "reg: 24  reads: 4570\n",
      "chrom 24 processed\n",
      "               WGM43          China_YR_MN mean depth:     0.019164 coverage: 1390\n",
      "histogram of read return codes\n",
      "read return code -166 ::        89\n",
      "read return code   -5 ::      3031\n",
      "read return code   -2 ::       200\n",
      "read return code   -1 ::        30\n",
      "read return code    1 ::      1220\n",
      "total: 4570\n",
      "\n",
      "## total reads: 4570  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 73680\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 1390 invalids 0\n",
      "\n",
      "##end of pulldown:        4.800 seconds cpu       47.649 Mbytes in use\n",
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/MZGM10-1\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/MZGM10-1\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: MZGM10-1\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/MZGM10-1.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: MZGM10-1\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/MZGM10-1.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 1302279325\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: MZGM10-1\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/MZGM10-1.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "./data/eigenstrat/y_snps_all2020_vO.snp: genetic distance set from physical distance\n",
      "number of valid snps: 73680\n",
      "sample MZGM10-1 is male\n",
      "numsnps: 73680  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for MZGM10-1 (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0        MZGM10-1  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/MZGM10-1.Y.bam Y \n",
      "triallelic: 36\n",
      "rawreads: 32359\n",
      "reg: 24  reads: 32359\n",
      "chrom 24 processed\n",
      "            MZGM10-1   China_Miaozigou_MN mean depth:     0.117902 coverage: 8088\n",
      "histogram of read return codes\n",
      "read return code -166 ::       396\n",
      "read return code   -5 ::     23754\n",
      "read return code   -2 ::       128\n",
      "read return code   -1 ::        75\n",
      "read return code    1 ::      8006\n",
      "total: 32359\n",
      "\n",
      "## total reads: 32359  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 73680\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 8088 invalids 0\n",
      "\n",
      "##end of pulldown:        4.930 seconds cpu       47.649 Mbytes in use\n",
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/MZGM16\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/MZGM16\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: MZGM16\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/MZGM16.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: MZGM16\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/MZGM16.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 833763199\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: MZGM16\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/MZGM16.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "./data/eigenstrat/y_snps_all2020_vO.snp: genetic distance set from physical distance\n",
      "number of valid snps: 73680\n",
      "sample MZGM16 is male\n",
      "numsnps: 73680  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for MZGM16 (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0          MZGM16  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/MZGM16.Y.bam Y \n",
      "triallelic: 2\n",
      "rawreads: 8394\n",
      "reg: 24  reads: 8394\n",
      "chrom 24 processed\n",
      "              MZGM16   China_Miaozigou_MN mean depth:     0.033917 coverage: 2447\n",
      "histogram of read return codes\n",
      "read return code -166 ::       149\n",
      "read return code   -5 ::      4947\n",
      "read return code   -2 ::      1162\n",
      "read return code   -1 ::        25\n",
      "read return code    1 ::      2111\n",
      "total: 8394\n",
      "\n",
      "## total reads: 8394  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 73680\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 2447 invalids 0\n",
      "\n",
      "##end of pulldown:        4.600 seconds cpu       47.649 Mbytes in use\n",
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/WGM94\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/WGM94\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: WGM94\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/WGM94.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: WGM94\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/WGM94.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 1770856057\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: WGM94\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/WGM94.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "./data/eigenstrat/y_snps_all2020_vO.snp: genetic distance set from physical distance\n",
      "number of valid snps: 73680\n",
      "sample WGM94 is male\n",
      "numsnps: 73680  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for WGM94 (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0           WGM94  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/WGM94.Y.bam Y \n",
      "triallelic: 111\n",
      "rawreads: 63057\n",
      "reg: 24  reads: 63057\n",
      "chrom 24 processed\n",
      "               WGM94          China_YR_MN mean depth:     0.264767 coverage: 16652\n",
      "histogram of read return codes\n",
      "read return code -166 ::      1532\n",
      "read return code   -5 ::     41787\n",
      "read return code   -2 ::      2477\n",
      "read return code   -1 ::       359\n",
      "read return code    1 ::     16902\n",
      "total: 63057\n",
      "\n",
      "## total reads: 63057  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 73680\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 16652 invalids 0\n",
      "\n",
      "##end of pulldown:        4.800 seconds cpu       47.649 Mbytes in use\n",
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/BLSM41\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/BLSM41\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: BLSM41\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/BLSM41.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: BLSM41\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/BLSM41.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 807055209\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: BLSM41\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/BLSM41.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "./data/eigenstrat/y_snps_all2020_vO.snp: genetic distance set from physical distance\n",
      "number of valid snps: 73680\n",
      "sample BLSM41 is male\n",
      "numsnps: 73680  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for BLSM41 (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0          BLSM41  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/BLSM41.Y.bam Y \n",
      "triallelic: 9\n",
      "rawreads: 3936\n",
      "reg: 24  reads: 3936\n",
      "chrom 24 processed\n",
      "              BLSM41         China_WLR_MN mean depth:     0.017454 coverage: 1264\n",
      "histogram of read return codes\n",
      "read return code -166 ::        64\n",
      "read return code   -5 ::      2689\n",
      "read return code   -2 ::        74\n",
      "read return code   -1 ::         8\n",
      "read return code    1 ::      1101\n",
      "total: 3936\n",
      "\n",
      "## total reads: 3936  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 73680\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 1264 invalids 0\n",
      "\n",
      "##end of pulldown:        4.620 seconds cpu       47.649 Mbytes in use\n",
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/SM-SGDLM6\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/SM-SGDLM6\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: SM-SGDLM6\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/SM-SGDLM6.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SM-SGDLM6\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/SM-SGDLM6.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 976348014\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: SM-SGDLM6\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/SM-SGDLM6.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "./data/eigenstrat/y_snps_all2020_vO.snp: genetic distance set from physical distance\n",
      "number of valid snps: 73680\n",
      "sample SM-SGDLM6 is male\n",
      "numsnps: 73680  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for SM-SGDLM6 (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0       SM-SGDLM6  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/SM-SGDLM6.Y.bam Y \n",
      "triallelic: 503\n",
      "rawreads: 384837\n",
      "reg: 24  reads: 384837\n",
      "chrom 24 processed\n",
      "           SM-SGDLM6      China_Shimao_LN mean depth:     1.649946 coverage: 57187\n",
      "histogram of read return codes\n",
      "read return code -166 ::      5854\n",
      "read return code   -5 ::    243545\n",
      "read return code   -2 ::     30484\n",
      "read return code   -1 ::      1524\n",
      "read return code    1 ::    103430\n",
      "total: 384837\n",
      "\n",
      "## total reads: 384837  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 73680\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 57185 invalids 0\n",
      "\n",
      "##end of pulldown:        5.990 seconds cpu       47.649 Mbytes in use\n",
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/SM-SGDLM27\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/SM-SGDLM27\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: SM-SGDLM27\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/SM-SGDLM27.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SM-SGDLM27\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/SM-SGDLM27.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 865099817\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: SM-SGDLM27\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/SM-SGDLM27.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "./data/eigenstrat/y_snps_all2020_vO.snp: genetic distance set from physical distance\n",
      "number of valid snps: 73680\n",
      "sample SM-SGDLM27 is male\n",
      "numsnps: 73680  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for SM-SGDLM27 (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0      SM-SGDLM27  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/SM-SGDLM27.Y.bam Y \n",
      "triallelic: 284\n",
      "rawreads: 258150\n",
      "reg: 24  reads: 258150\n",
      "chrom 24 processed\n",
      "          SM-SGDLM27      China_Shimao_LN mean depth:     1.105049 coverage: 42605\n",
      "histogram of read return codes\n",
      "read return code -166 ::      3737\n",
      "read return code   -5 ::    176095\n",
      "read return code   -2 ::      6687\n",
      "read return code   -1 ::       745\n",
      "read return code    1 ::     70886\n",
      "total: 258150\n",
      "\n",
      "## total reads: 258150  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 73680\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 42605 invalids 0\n",
      "\n",
      "##end of pulldown:        5.470 seconds cpu       47.649 Mbytes in use\n",
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/WD-WT1H16\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/WD-WT1H16\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: WD-WT1H16\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/WD-WT1H16.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: WD-WT1H16\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/WD-WT1H16.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 700465745\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: WD-WT1H16\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/WD-WT1H16.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "./data/eigenstrat/y_snps_all2020_vO.snp: genetic distance set from physical distance\n",
      "number of valid snps: 73680\n",
      "sample WD-WT1H16 is male\n",
      "numsnps: 73680  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for WD-WT1H16 (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0       WD-WT1H16  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/WD-WT1H16.Y.bam Y \n",
      "triallelic: 711\n",
      "rawreads: 548903\n",
      "reg: 24  reads: 548903\n",
      "chrom 24 processed\n",
      "           WD-WT1H16          China_YR_LN mean depth:     2.473969 coverage: 62517\n",
      "histogram of read return codes\n",
      "read return code -166 ::     11796\n",
      "read return code   -5 ::    318596\n",
      "read return code   -2 ::     66385\n",
      "read return code   -1 ::      2017\n",
      "read return code    1 ::    150109\n",
      "total: 548903\n",
      "\n",
      "## total reads: 548903  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 73680\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 62515 invalids 0\n",
      "\n",
      "##end of pulldown:        6.760 seconds cpu       47.649 Mbytes in use\n",
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/LJM3\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/LJM3\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: LJM3\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/LJM3.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: LJM3\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/LJM3.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 201967191\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: LJM3\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/LJM3.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "./data/eigenstrat/y_snps_all2020_vO.snp: genetic distance set from physical distance\n",
      "number of valid snps: 73680\n",
      "sample LJM3 is male\n",
      "numsnps: 73680  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for LJM3 (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0            LJM3  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/LJM3.Y.bam Y \n",
      "triallelic: 93\n",
      "rawreads: 156467\n",
      "reg: 24  reads: 156467\n",
      "chrom 24 processed\n",
      "                LJM3    China_Upper_YR_LN mean depth:     0.537215 coverage: 29881\n",
      "histogram of read return codes\n",
      "read return code -166 ::      3492\n",
      "read return code   -5 ::     69198\n",
      "read return code   -2 ::     50775\n",
      "read return code   -1 ::       548\n",
      "read return code    1 ::     32454\n",
      "total: 156467\n",
      "\n",
      "## total reads: 156467  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 73680\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 29881 invalids 0\n",
      "\n",
      "##end of pulldown:        5.220 seconds cpu       47.649 Mbytes in use\n",
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/PLTM311\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/PLTM311\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: PLTM311\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/PLTM311.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: PLTM311\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/PLTM311.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 724208282\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: PLTM311\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/PLTM311.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "./data/eigenstrat/y_snps_all2020_vO.snp: genetic distance set from physical distance\n",
      "number of valid snps: 73680\n",
      "sample PLTM311 is male\n",
      "numsnps: 73680  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for PLTM311 (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0         PLTM311  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/PLTM311.Y.bam Y \n",
      "triallelic: 382\n",
      "rawreads: 286774\n",
      "reg: 24  reads: 286774\n",
      "chrom 24 processed\n",
      "             PLTM311          China_YR_LN mean depth:     1.279045 coverage: 49147\n",
      "histogram of read return codes\n",
      "read return code -166 ::      5954\n",
      "read return code   -5 ::    169095\n",
      "read return code   -2 ::     32213\n",
      "read return code   -1 ::       980\n",
      "read return code    1 ::     78532\n",
      "total: 286774\n",
      "\n",
      "## total reads: 286774  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 73680\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 49145 invalids 0\n",
      "\n",
      "##end of pulldown:        5.670 seconds cpu       47.649 Mbytes in use\n",
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/PLTM312\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/PLTM312\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: PLTM312\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/PLTM312.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: PLTM312\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/PLTM312.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 763655526\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: PLTM312\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/PLTM312.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "./data/eigenstrat/y_snps_all2020_vO.snp: genetic distance set from physical distance\n",
      "number of valid snps: 73680\n",
      "sample PLTM312 is male\n",
      "numsnps: 73680  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for PLTM312 (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0         PLTM312  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/PLTM312.Y.bam Y \n",
      "triallelic: 25\n",
      "rawreads: 15605\n",
      "reg: 24  reads: 15605\n",
      "chrom 24 processed\n",
      "             PLTM312 China_YR_LN_1d.or.2d.rel.PLTM311 mean depth:     0.071621 coverage: 4992\n",
      "histogram of read return codes\n",
      "read return code -166 ::       296\n",
      "read return code   -5 ::     10206\n",
      "read return code   -2 ::       556\n",
      "read return code   -1 ::        41\n",
      "read return code    1 ::      4506\n",
      "total: 15605\n",
      "\n",
      "## total reads: 15605  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 73680\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 4992 invalids 0\n",
      "\n",
      "##end of pulldown:        4.610 seconds cpu       47.649 Mbytes in use\n",
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/HJTM109\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/HJTM109\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: HJTM109\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/HJTM109.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: HJTM109\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/HJTM109.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 427573309\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: HJTM109\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/HJTM109.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "./data/eigenstrat/y_snps_all2020_vO.snp: genetic distance set from physical distance\n",
      "number of valid snps: 73680\n",
      "sample HJTM109 is male\n",
      "numsnps: 73680  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for HJTM109 (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0         HJTM109  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/HJTM109.Y.bam Y \n",
      "triallelic: 145\n",
      "rawreads: 96980\n",
      "reg: 24  reads: 96980\n",
      "chrom 24 processed\n",
      "             HJTM109          China_YR_LN mean depth:     0.353787 coverage: 20593\n",
      "histogram of read return codes\n",
      "read return code -166 ::      1578\n",
      "read return code   -5 ::     71032\n",
      "read return code   -2 ::       153\n",
      "read return code   -1 ::       281\n",
      "read return code    1 ::     23936\n",
      "total: 96980\n",
      "\n",
      "## total reads: 96980  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 73680\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 20593 invalids 0\n",
      "\n",
      "##end of pulldown:        4.880 seconds cpu       47.649 Mbytes in use\n",
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/LJM5\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/LJM5\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: LJM5\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/LJM5.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: LJM5\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/LJM5.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 1074513495\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: LJM5\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/LJM5.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "./data/eigenstrat/y_snps_all2020_vO.snp: genetic distance set from physical distance\n",
      "number of valid snps: 73680\n",
      "sample LJM5 is male\n",
      "numsnps: 73680  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for LJM5 (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0            LJM5  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/LJM5.Y.bam Y \n",
      "triallelic: 137\n",
      "rawreads: 89450\n",
      "reg: 24  reads: 89450\n",
      "chrom 24 processed\n",
      "                LJM5 China_Upper_YR_LN_father.or.son.LJM3 mean depth:     0.390662 coverage: 23304\n",
      "histogram of read return codes\n",
      "read return code -166 ::      1597\n",
      "read return code   -5 ::     57899\n",
      "read return code   -2 ::      4953\n",
      "read return code   -1 ::       400\n",
      "read return code    1 ::     24601\n",
      "total: 89450\n",
      "\n",
      "## total reads: 89450  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 73680\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 23304 invalids 0\n",
      "\n",
      "##end of pulldown:        4.920 seconds cpu       47.649 Mbytes in use\n",
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/LJM25\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/LJM25\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: LJM25\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/LJM25.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: LJM25\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/LJM25.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 119626014\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: LJM25\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/LJM25.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "./data/eigenstrat/y_snps_all2020_vO.snp: genetic distance set from physical distance\n",
      "number of valid snps: 73680\n",
      "sample LJM25 is male\n",
      "numsnps: 73680  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for LJM25 (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0           LJM25  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/LJM25.Y.bam Y \n",
      "triallelic: 9\n",
      "rawreads: 4465\n",
      "reg: 24  reads: 4465\n",
      "chrom 24 processed\n",
      "               LJM25    China_Upper_YR_LN mean depth:     0.019015 coverage: 1385\n",
      "histogram of read return codes\n",
      "read return code -166 ::       107\n",
      "read return code   -5 ::      2918\n",
      "read return code   -2 ::       204\n",
      "read return code   -1 ::        32\n",
      "read return code    1 ::      1204\n",
      "total: 4465\n",
      "\n",
      "## total reads: 4465  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 73680\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 1385 invalids 0\n",
      "\n",
      "##end of pulldown:        4.580 seconds cpu       47.649 Mbytes in use\n",
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/LJM14\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/LJM14\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: LJM14\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/LJM14.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: LJM14\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/LJM14.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 852577238\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: LJM14\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/LJM14.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "./data/eigenstrat/y_snps_all2020_vO.snp: genetic distance set from physical distance\n",
      "number of valid snps: 73680\n",
      "sample LJM14 is male\n",
      "numsnps: 73680  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for LJM14 (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0           LJM14  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/LJM14.Y.bam Y \n",
      "triallelic: 238\n",
      "rawreads: 131035\n",
      "reg: 24  reads: 131035\n",
      "chrom 24 processed\n",
      "               LJM14    China_Upper_YR_LN mean depth:     0.536889 coverage: 29294\n",
      "histogram of read return codes\n",
      "read return code -166 ::      2854\n",
      "read return code   -5 ::     86119\n",
      "read return code   -2 ::      6811\n",
      "read return code   -1 ::       842\n",
      "read return code    1 ::     34409\n",
      "total: 131035\n",
      "\n",
      "## total reads: 131035  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 73680\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 29294 invalids 0\n",
      "\n",
      "##end of pulldown:        5.030 seconds cpu       47.649 Mbytes in use\n",
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/HJTM107\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/HJTM107\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: HJTM107\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/HJTM107.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: HJTM107\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/HJTM107.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 1369202740\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: HJTM107\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/HJTM107.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "./data/eigenstrat/y_snps_all2020_vO.snp: genetic distance set from physical distance\n",
      "number of valid snps: 73680\n",
      "sample HJTM107 is male\n",
      "numsnps: 73680  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for HJTM107 (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0         HJTM107  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/HJTM107.Y.bam Y \n",
      "triallelic: 185\n",
      "rawreads: 121100\n",
      "reg: 24  reads: 121100\n",
      "chrom 24 processed\n",
      "             HJTM107          China_YR_LN mean depth:     0.469870 coverage: 25509\n",
      "histogram of read return codes\n",
      "read return code -166 ::      1636\n",
      "read return code   -5 ::     86520\n",
      "read return code   -2 ::      1377\n",
      "read return code   -1 ::       419\n",
      "read return code    1 ::     31148\n",
      "total: 121100\n",
      "\n",
      "## total reads: 121100  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 73680\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 25509 invalids 0\n",
      "\n",
      "##end of pulldown:        4.960 seconds cpu       47.649 Mbytes in use\n",
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/JCKM1-1\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/JCKM1-1\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: JCKM1-1\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/JCKM1-1.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: JCKM1-1\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/JCKM1-1.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 2030663831\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: JCKM1-1\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/JCKM1-1.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "./data/eigenstrat/y_snps_all2020_vO.snp: genetic distance set from physical distance\n",
      "number of valid snps: 73680\n",
      "sample JCKM1-1 is male\n",
      "numsnps: 73680  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for JCKM1-1 (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0         JCKM1-1  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/JCKM1-1.Y.bam Y \n",
      "triallelic: 41\n",
      "rawreads: 93062\n",
      "reg: 24  reads: 93062\n",
      "chrom 24 processed\n",
      "             JCKM1-1    China_Upper_YR_LN mean depth:     0.284826 coverage: 18005\n",
      "histogram of read return codes\n",
      "read return code -166 ::      2351\n",
      "read return code   -5 ::     33264\n",
      "read return code   -2 ::     40249\n",
      "read return code   -1 ::       325\n",
      "read return code    1 ::     16873\n",
      "total: 93062\n",
      "\n",
      "## total reads: 93062  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 73680\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 18005 invalids 0\n",
      "\n",
      "##end of pulldown:        4.970 seconds cpu       47.649 Mbytes in use\n",
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/EDM176\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/EDM176\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: EDM176\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/EDM176.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: EDM176\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/EDM176.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 173715674\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: EDM176\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/EDM176.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "./data/eigenstrat/y_snps_all2020_vO.snp: genetic distance set from physical distance\n",
      "number of valid snps: 73680\n",
      "sample EDM176 is male\n",
      "numsnps: 73680  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for EDM176 (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0          EDM176  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/EDM176.Y.bam Y \n",
      "triallelic: 290\n",
      "rawreads: 189375\n",
      "reg: 24  reads: 189375\n",
      "chrom 24 processed\n",
      "              EDM176         China_WLR_LN mean depth:     0.839468 coverage: 35880\n",
      "histogram of read return codes\n",
      "read return code -166 ::      4311\n",
      "read return code   -5 ::    125366\n",
      "read return code   -2 ::      5450\n",
      "read return code   -1 ::      1119\n",
      "read return code    1 ::     53129\n",
      "total: 189375\n",
      "\n",
      "## total reads: 189375  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 73680\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 35879 invalids 0\n",
      "\n",
      "##end of pulldown:        5.200 seconds cpu       47.649 Mbytes in use\n",
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/EDM124\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/EDM124\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: EDM124\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/EDM124.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: EDM124\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/EDM124.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 1985023902\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: EDM124\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/EDM124.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "./data/eigenstrat/y_snps_all2020_vO.snp: genetic distance set from physical distance\n",
      "number of valid snps: 73680\n",
      "sample EDM124 is male\n",
      "numsnps: 73680  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for EDM124 (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0          EDM124  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/EDM124.Y.bam Y \n",
      "triallelic: 19\n",
      "rawreads: 16792\n",
      "reg: 24  reads: 16792\n",
      "chrom 24 processed\n",
      "              EDM124         China_WLR_LN mean depth:     0.069123 coverage: 4833\n",
      "histogram of read return codes\n",
      "read return code -166 ::       379\n",
      "read return code   -5 ::     10072\n",
      "read return code   -2 ::      1889\n",
      "read return code   -1 ::        95\n",
      "read return code    1 ::      4357\n",
      "total: 16792\n",
      "\n",
      "## total reads: 16792  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 73680\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 4833 invalids 0\n",
      "\n",
      "##end of pulldown:        4.630 seconds cpu       47.649 Mbytes in use\n",
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/JXNTM2\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/JXNTM2\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: JXNTM2\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/JXNTM2.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: JXNTM2\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/JXNTM2.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 1391352139\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: JXNTM2\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/JXNTM2.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "./data/eigenstrat/y_snps_all2020_vO.snp: genetic distance set from physical distance\n",
      "number of valid snps: 73680\n",
      "sample JXNTM2 is male\n",
      "numsnps: 73680  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for JXNTM2 (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0          JXNTM2  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/JXNTM2.Y.bam Y \n",
      "triallelic: 258\n",
      "rawreads: 179608\n",
      "reg: 24  reads: 179608\n",
      "chrom 24 processed\n",
      "              JXNTM2        China_YR_LBIA mean depth:     0.767983 coverage: 37284\n",
      "histogram of read return codes\n",
      "read return code -166 ::      3017\n",
      "read return code   -5 ::    119921\n",
      "read return code   -2 ::      7217\n",
      "read return code   -1 ::       603\n",
      "read return code    1 ::     48850\n",
      "total: 179608\n",
      "\n",
      "## total reads: 179608  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 73680\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 37284 invalids 0\n",
      "\n",
      "##end of pulldown:        5.170 seconds cpu       47.649 Mbytes in use\n",
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/91KLH18\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/91KLH18\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: 91KLH18\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/91KLH18.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: 91KLH18\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/91KLH18.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 750727689\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: 91KLH18\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/91KLH18.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "./data/eigenstrat/y_snps_all2020_vO.snp: genetic distance set from physical distance\n",
      "number of valid snps: 73680\n",
      "sample 91KLH18 is male\n",
      "numsnps: 73680  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for 91KLH18 (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0         91KLH18  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/91KLH18.Y.bam Y \n",
      "triallelic: 27\n",
      "rawreads: 44839\n",
      "reg: 24  reads: 44839\n",
      "chrom 24 processed\n",
      "             91KLH18         China_WLR_BA mean depth:     0.177633 coverage: 11771\n",
      "histogram of read return codes\n",
      "read return code -166 ::       870\n",
      "read return code   -5 ::     26035\n",
      "read return code   -2 ::      6638\n",
      "read return code   -1 ::       116\n",
      "read return code    1 ::     11180\n",
      "total: 44839\n",
      "\n",
      "## total reads: 44839  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 73680\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 11771 invalids 0\n",
      "\n",
      "##end of pulldown:        4.730 seconds cpu       47.649 Mbytes in use\n",
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/91KLM2\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/91KLM2\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: 91KLM2\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/91KLM2.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: 91KLM2\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/91KLM2.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 1961362920\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: 91KLM2\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/91KLM2.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "./data/eigenstrat/y_snps_all2020_vO.snp: genetic distance set from physical distance\n",
      "number of valid snps: 73680\n",
      "sample 91KLM2 is male\n",
      "numsnps: 73680  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for 91KLM2 (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0          91KLM2  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/91KLM2.Y.bam Y \n",
      "triallelic: 60\n",
      "rawreads: 130229\n",
      "reg: 24  reads: 130229\n",
      "chrom 24 processed\n",
      "              91KLM2       China_WLR_BA_o mean depth:     0.532519 coverage: 29316\n",
      "histogram of read return codes\n",
      "read return code -166 ::      1824\n",
      "read return code   -5 ::     81411\n",
      "read return code   -2 ::     12997\n",
      "read return code   -1 ::       351\n",
      "read return code    1 ::     33646\n",
      "total: 130229\n",
      "\n",
      "## total reads: 130229  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 73680\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 29316 invalids 0\n",
      "\n",
      "##end of pulldown:        5.030 seconds cpu       47.649 Mbytes in use\n",
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/HJTW13\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/HJTW13\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: HJTW13\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/HJTW13.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: HJTW13\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/HJTW13.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 642312081\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: HJTW13\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/HJTW13.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "./data/eigenstrat/y_snps_all2020_vO.snp: genetic distance set from physical distance\n",
      "number of valid snps: 73680\n",
      "sample HJTW13 is male\n",
      "numsnps: 73680  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for HJTW13 (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0          HJTW13  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/HJTW13.Y.bam Y \n",
      "triallelic: 5\n",
      "rawreads: 2582\n",
      "reg: 24  reads: 2582\n",
      "chrom 24 processed\n",
      "              HJTW13        China_YR_LBIA mean depth:     0.011876 coverage: 871\n",
      "histogram of read return codes\n",
      "read return code -166 ::        47\n",
      "read return code   -5 ::      1598\n",
      "read return code   -2 ::       191\n",
      "read return code   -1 ::         9\n",
      "read return code    1 ::       737\n",
      "total: 2582\n",
      "\n",
      "## total reads: 2582  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 73680\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 871 invalids 0\n",
      "\n",
      "##end of pulldown:        4.560 seconds cpu       47.649 Mbytes in use\n",
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/LGM41\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/LGM41\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: LGM41\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/LGM41.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: LGM41\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/LGM41.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 1726162059\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: LGM41\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/LGM41.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "./data/eigenstrat/y_snps_all2020_vO.snp: genetic distance set from physical distance\n",
      "number of valid snps: 73680\n",
      "sample LGM41 is male\n",
      "numsnps: 73680  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for LGM41 (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0           LGM41  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/LGM41.Y.bam Y \n",
      "triallelic: 188\n",
      "rawreads: 142777\n",
      "reg: 24  reads: 142777\n",
      "chrom 24 processed\n",
      "               LGM41        China_YR_LBIA mean depth:     0.606338 coverage: 32004\n",
      "histogram of read return codes\n",
      "read return code -166 ::      2178\n",
      "read return code   -5 ::     92024\n",
      "read return code   -2 ::      9594\n",
      "read return code   -1 ::       546\n",
      "read return code    1 ::     38435\n",
      "total: 142777\n",
      "\n",
      "## total reads: 142777  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 73680\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 32004 invalids 0\n",
      "\n",
      "##end of pulldown:        5.060 seconds cpu       47.649 Mbytes in use\n",
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/DCZ-M21II\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/DCZ-M21II\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: DCZ-M21II\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/DCZ-M21II.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: DCZ-M21II\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/DCZ-M21II.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 690521528\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: DCZ-M21II\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/DCZ-M21II.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "./data/eigenstrat/y_snps_all2020_vO.snp: genetic distance set from physical distance\n",
      "number of valid snps: 73680\n",
      "sample DCZ-M21II is male\n",
      "numsnps: 73680  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for DCZ-M21II (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0       DCZ-M21II  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/DCZ-M21II.Y.bam Y \n",
      "triallelic: 30\n",
      "rawreads: 79098\n",
      "reg: 24  reads: 79098\n",
      "chrom 24 processed\n",
      "           DCZ-M21II    China_Upper_YR_IA mean depth:     0.318078 coverage: 19588\n",
      "histogram of read return codes\n",
      "read return code -166 ::      1468\n",
      "read return code   -5 ::     37288\n",
      "read return code   -2 ::     21312\n",
      "read return code   -1 ::       229\n",
      "read return code    1 ::     18801\n",
      "total: 79098\n",
      "\n",
      "## total reads: 79098  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 73680\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 19588 invalids 0\n",
      "\n",
      "##end of pulldown:        4.890 seconds cpu       47.649 Mbytes in use\n",
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/ZLNR-1\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/ZLNR-1\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: ZLNR-1\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/ZLNR-1.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: ZLNR-1\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/ZLNR-1.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 1278881979\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: ZLNR-1\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/ZLNR-1.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "./data/eigenstrat/y_snps_all2020_vO.snp: genetic distance set from physical distance\n",
      "number of valid snps: 73680\n",
      "sample ZLNR-1 is male\n",
      "numsnps: 73680  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for ZLNR-1 (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0          ZLNR-1  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/ZLNR-1.Y.bam Y \n",
      "triallelic: 1\n",
      "rawreads: 4575\n",
      "reg: 24  reads: 4575\n",
      "chrom 24 processed\n",
      "              ZLNR-1          China_AR_IA mean depth:     0.020046 coverage: 1455\n",
      "histogram of read return codes\n",
      "read return code -166 ::        71\n",
      "read return code   -5 ::      2536\n",
      "read return code   -2 ::       736\n",
      "read return code   -1 ::        26\n",
      "read return code    1 ::      1206\n",
      "total: 4575\n",
      "\n",
      "## total reads: 4575  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 73680\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 1455 invalids 0\n",
      "\n",
      "##end of pulldown:        6.230 seconds cpu       47.649 Mbytes in use\n",
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/MGS-M7L\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/MGS-M7L\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: MGS-M7L\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/MGS-M7L.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: MGS-M7L\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/MGS-M7L.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 142416265\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: MGS-M7L\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/MGS-M7L.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "./data/eigenstrat/y_snps_all2020_vO.snp: genetic distance set from physical distance\n",
      "number of valid snps: 73680\n",
      "sample MGS-M7L is male\n",
      "numsnps: 73680  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for MGS-M7L (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0         MGS-M7L  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/MGS-M7L.Y.bam Y \n",
      "triallelic: 1\n",
      "rawreads: 5446\n",
      "reg: 24  reads: 5446\n",
      "chrom 24 processed\n",
      "             MGS-M7L China_AR_Xianbei_IA_father.or.son.MGS-M7R mean depth:     0.016490 coverage: 1199\n",
      "histogram of read return codes\n",
      "read return code -166 ::       142\n",
      "read return code   -5 ::      1860\n",
      "read return code   -2 ::      2459\n",
      "read return code   -1 ::        19\n",
      "read return code    1 ::       966\n",
      "total: 5446\n",
      "\n",
      "## total reads: 5446  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 73680\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 1199 invalids 0\n",
      "\n",
      "##end of pulldown:        4.600 seconds cpu       47.649 Mbytes in use\n",
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/MGS-M7R\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/MGS-M7R\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: MGS-M7R\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/MGS-M7R.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: MGS-M7R\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/MGS-M7R.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 954575829\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: MGS-M7R\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/MGS-M7R.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "./data/eigenstrat/y_snps_all2020_vO.snp: genetic distance set from physical distance\n",
      "number of valid snps: 73680\n",
      "sample MGS-M7R is male\n",
      "numsnps: 73680  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for MGS-M7R (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0         MGS-M7R  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/MGS-M7R.Y.bam Y \n",
      "triallelic: 4\n",
      "rawreads: 13971\n",
      "reg: 24  reads: 13971\n",
      "chrom 24 processed\n",
      "             MGS-M7R  China_AR_Xianbei_IA mean depth:     0.039007 coverage: 2796\n",
      "histogram of read return codes\n",
      "read return code -166 ::       331\n",
      "read return code   -5 ::      4448\n",
      "read return code   -2 ::      6845\n",
      "read return code   -1 ::        44\n",
      "read return code    1 ::      2303\n",
      "total: 13971\n",
      "\n",
      "## total reads: 13971  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 73680\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 2796 invalids 0\n",
      "\n",
      "##end of pulldown:        4.630 seconds cpu       47.649 Mbytes in use\n",
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/MGS-M6\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/MGS-M6\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: MGS-M6\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/MGS-M6.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: MGS-M6\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/MGS-M6.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 1565007350\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: MGS-M6\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: ./data/eigenstrat/y_snps_all2020_vO.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/MGS-M6.Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "./data/eigenstrat/y_snps_all2020_vO.snp: genetic distance set from physical distance\n",
      "number of valid snps: 73680\n",
      "sample MGS-M6 is male\n",
      "numsnps: 73680  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for MGS-M6 (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0          MGS-M6  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/A-raw/B-chrY/MGS-M6.Y.bam Y \n",
      "triallelic: 4\n",
      "rawreads: 6669\n",
      "reg: 24  reads: 6669\n",
      "chrom 24 processed\n",
      "              MGS-M6  China_AR_Xianbei_IA mean depth:     0.023412 coverage: 1705\n",
      "histogram of read return codes\n",
      "read return code -166 ::       160\n",
      "read return code   -5 ::      2757\n",
      "read return code   -2 ::      2352\n",
      "read return code   -1 ::        21\n",
      "read return code    1 ::      1379\n",
      "total: 6669\n",
      "\n",
      "## total reads: 6669  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 73680\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 1705 invalids 0\n",
      "\n",
      "##end of pulldown:        4.580 seconds cpu       47.649 Mbytes in use\n",
      "CPU times: user 4.32 s, sys: 1.18 s, total: 5.5 s\n",
      "Wall time: 2min 35s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "### Create a parfile to run\n",
    "for iid in iids:\n",
    "    ### Produce the Y chromosome\n",
    "    savepath = create_parfile_from_ref(iid = iid,  #I10266\n",
    "                                       bam_path = bam_folder + iid + \".Y.bam\",\n",
    "                                       outfolder = \"/n/groups/reich/hringbauer/git/tibet_aDNA/Data/Y_pulldowns/\",\n",
    "                                       snp_file=\"./data/eigenstrat/y_snps_all2020_vO.snp\")\n",
    "    run_pulldown_y(path_parfile=savepath) # Run the Pulldown"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/ZLNR-2\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/ZLNR-2\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: ZLNR-2\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: /n/groups/reich/matt/pipeline/static/1240kSNP.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/B-fix/ZLNR-2/hg19/aln.mapped.rmdupse.md_no1kgvar.rg.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: ZLNR-2\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: ./output/ning20/ZLNR-2.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 1695689265\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: ZLNR-2\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: /n/groups/reich/matt/pipeline/static/1240kSNP.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/B-fix/ZLNR-2/hg19/aln.mapped.rmdupse.md_no1kgvar.rg.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: ./output/ning20/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "number of valid snps: 32670\n",
      "numsnps: 1233013  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for ZLNR-2 (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0          ZLNR-2  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/B-fix/ZLNR-2/hg19/aln.mapped.rmdupse.md_no1kgvar.rg.bam Y \n",
      "triallelic: 0\n",
      "rawreads: 0\n",
      "*** no data... database problem? ***\n",
      "reg: 24  reads: 0\n",
      "chrom 24 processed\n",
      "              ZLNR-2          China_AR_EN mean depth:     0.000000 coverage: 0\n",
      "histogram of read return codes\n",
      "total: 0\n",
      "\n",
      "## total reads: 0  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 1233013\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 0 invalids 0\n",
      "\n",
      "##end of pulldown:        8.310 seconds cpu      759.505 Mbytes in use\n",
      "CPU times: user 205 ms, sys: 58.7 ms, total: 264 ms\n",
      "Wall time: 8.62 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "par_dc = give_default_par_dc()\n",
    "### Create a parfile to run\n",
    "for iid in iids[:1]:\n",
    "    ### Produce the Y chromosome\n",
    "    savepath = create_parfile_from_ref(df=df, \n",
    "                                       iid=iid,  #I10266\n",
    "                                       par_dc=par_dc,\n",
    "                                       outfolder=\"./output/ning20/\")\n",
    "    run_pulldown_y(path_parfile=savepath) # Run the Pulldown"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Run the Full ISOGG SNP set "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../*Y.bam\n",
      "Successfully saved to /n/groups/reich/hringbauer/git/y_chrom/parfiles/Loschbour_published.DG\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/Loschbour_published.DG\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: Loschbour_published.DG\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/hringbauer/explore_ntbk/v42/V42.3/v42.3.ind\n",
      "snpname: /n/groups/reich/hringbauer/git/y_chrom/data/eigenstrat/y_snps_all2020.snp\n",
      "bamname: ../*Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: Loschbour_published.DG\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/y_chrom/output/pulldowns_all/Loschbour_published.DG.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 1354517211\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: Loschbour_published.DG\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/hringbauer/explore_ntbk/v42/V42.3/v42.3.ind\n",
      "snpname: /n/groups/reich/hringbauer/git/y_chrom/data/eigenstrat/y_snps_all2020.snp\n",
      "bamname: ../*Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/y_chrom/output/pulldowns_all/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "/n/groups/reich/hringbauer/git/y_chrom/data/eigenstrat/y_snps_all2020.snp: genetic distance set from physical distance\n",
      "number of valid snps: 72933\n",
      "sample hed.DG is male\n",
      "numsnps: 72933  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for Loschbour_published.DG (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0 Loschbour_published.DG  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view ../*Y.bam Y \n",
      "[E::hts_open_format] Failed to open file ../*Y.bam\n",
      "samtools view: failed to open \"../*Y.bam\" for reading: No such file or directory\n",
      "triallelic: 0\n",
      "rawreads: 0\n",
      "*** no data... database problem? ***\n",
      "reg: 24  reads: 0\n",
      "chrom 24 processed\n",
      "Loschbour_published.DG Luxembourg_Loschbour_published.DG mean depth:     0.000000 coverage: 0\n",
      "histogram of read return codes\n",
      "total: 0\n",
      "\n",
      "## total reads: 0  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 72933\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 0 invalids 0\n",
      "\n",
      "##end of pulldown:        8.170 seconds cpu       46.801 Mbytes in use\n",
      "CPU times: user 187 ms, sys: 38.3 ms, total: 225 ms\n",
      "Wall time: 8.53 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "### Create a parfile to run\n",
    "savepath = create_parfile_from_ref(df=df, \n",
    "                                   iid=\"Loschbour_published.DG\",  #I10266\n",
    "                                   par_dc=par_dc,\n",
    "                                   \n",
    "                                   use_fullisogg=True)\n",
    "### Produce Lots of Output (toggled)\n",
    "run_pulldown_y(path_parfile=savepath)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Area 51"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Print old Parfile\n",
    "path = df1['Data: pulldown logfile location'].values[0]\n",
    "file = open(path,mode='r')\n",
    "txt = file.read()\n",
    "print(txt)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
