{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "compute-e-16-237.o2.rc.hms.harvard.edu\n",
      "HSM Computational partition detected.\n",
      "/n/groups/reich/hringbauer/git/y_chrom\n",
      "CPU Count: 28\n",
      "3.7.4 (default, Sep 11 2019, 11:24:51) \n",
      "[GCC 6.2.0]\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "import os  # For Saving to Folder\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "import socket\n",
    "import os as os\n",
    "import sys as sys\n",
    "import multiprocessing as mp\n",
    "\n",
    "### For Arial Font\n",
    "from matplotlib import rcParams\n",
    "rcParams['font.family'] = 'sans-serif'   # Set the defaul\n",
    "### Make sure to have the font installed (it is on cluster for Harald)\n",
    "rcParams['font.sans-serif'] = ['Arial']\n",
    "\n",
    "socket_name = socket.gethostname()\n",
    "print(socket_name)\n",
    "\n",
    "if socket_name.startswith(\"compute-\"):\n",
    "    print(\"HSM Computational partition detected.\")\n",
    "    path = \"/n/groups/reich/hringbauer/git/y_chrom/\"  # The Path on Midway Cluster\n",
    "else:\n",
    "    raise RuntimeWarning(\"Not compatible machine. Check!!\")\n",
    "\n",
    "os.chdir(path)  # Set the right Path (in line with Atom default)\n",
    "# Show the current working directory. Should be HAPSBURG/Notebooks/ParallelRuns\n",
    "print(os.getcwd())\n",
    "print(f\"CPU Count: {mp.cpu_count()}\")\n",
    "print(sys.version)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Define Parameters for param file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_parfile_from_ref(df, iid,\n",
    "                            par_dc, \n",
    "                            savepath = \"/n/groups/reich/hringbauer/git/y_chrom/parfiles/\",\n",
    "                            outfolder=\"\",\n",
    "                            col_id = 'Version ID',\n",
    "                            use_fullisogg=False):\n",
    "    \"\"\"Create a new parfile from reference dataframe df\n",
    "    iid: What individual to prepare\n",
    "    id_col: What column to use in anno file for IID\n",
    "    savepath: Path to save the new parfile to\n",
    "    outfolder: p\"\"\"\n",
    "    bam_path = extram_bam_df(df, iid=iid, col_id=col_id)\n",
    "    \n",
    "    if use_fullisogg:\n",
    "        par_dc[\"snpname\"] =  \"/n/groups/reich/hringbauer/git/y_chrom/data/eigenstrat/y_snps_all2020.snp\"\n",
    "        par_dc[\"readdepthname\"] = '/n/groups/reich/hringbauer/git/y_chrom/output/pulldowns_all/SAMP.txt'\n",
    "    \n",
    "    if os.path.isdir(bam_path):\n",
    "        bam_path = os.path.join(bam_path, \"*Y.bam\")  # Append the Y BAM\n",
    "        print(bam_path)\n",
    "    #else:\n",
    "        #bam_path = bam_path + \".by_chr/*Y.bam\"   # Hack for Ning samples. Not clear how general\n",
    "        #print(\"Warning. Ning2020 hack run. General?\")\n",
    "    savepath = create_parfile(par_dc, sample=iid, bam_path=bam_path,\n",
    "                   savepath=savepath, outfolder=outfolder)\n",
    "    return savepath\n",
    "\n",
    "def create_parfile(par_dc, sample=\"test\", bam_path=\"\", outfolder=\"\",\n",
    "                   savepath=\"/n/groups/reich/hringbauer/git/y_chrom/parfiles/\"):\n",
    "    \"\"\"Creates parfile from dictonary of Parameters\n",
    "    savepath: Where to save the parifle to\"\"\"\n",
    "    ### Modify Parameter Dictionary\n",
    "    par_dc[\"SAMP\"] = sample\n",
    "    par_dc[\"bamname\"] = bam_path\n",
    "    \n",
    "    if len(outfolder)>0:\n",
    "        par_dc[\"readdepthname\"] = os.path.join(outfolder ,\"SAMP.txt\")\n",
    "        \n",
    "    \n",
    "    ### Save to File\n",
    "    savepath = os.path.join(savepath,sample)\n",
    "    with open(savepath,\"w\") as f:\n",
    "        for k, v in par_dc.items():        \n",
    "            f.write(f\"{k}: {v}\\n\")\n",
    "    print(f\"Successfully saved parfile to {savepath}\")\n",
    "    return savepath\n",
    "    \n",
    "def run_pulldown_y(path_parfile, path_bin = \"./scripts/pulldown\", output=True):\n",
    "    \"\"\"Run Pulldown via par file.\n",
    "    path_parfile: Which parfile to run\n",
    "    path_bin: Where to find the binary to run it\"\"\"\n",
    "    if output:\n",
    "        ! $path_bin -p $path_parfile\n",
    "    else:\n",
    "        ! $path_bin -p $path_parfile  > /dev/null\n",
    "    \n",
    "def extram_bam_df(df, iid,\n",
    "                  col_id = 'Instance ID (\"_all\" means includes a mix of UDG-treated and non-UDG-treated; \"_published\" distinguishes a published sample for a still-unpublished higher quality version)', \n",
    "                  col_bam = 'Data: autosomal bam'):\n",
    "    \"\"\"Return bam Path from Meta File\"\"\"\n",
    "    idx = (df[col_id]==iid)\n",
    "    assert(np.sum(idx)>0) \n",
    "    bam_path = df.loc[idx, col_bam].values[0]\n",
    "    return bam_path\n",
    "\n",
    "def give_default_par_dc():\n",
    "    \"\"\"Return the default parameter dictionary\"\"\"\n",
    "    par_dc = {\"BASE\": \"/n/groups/reich/matt/pipeline/static\",\n",
    "          \"SAMP\": \"FILL_IN\",\n",
    "          \"D2\": \"./output\",\n",
    "          \"indivname\": \"/n/groups/reich/DAVID/V43/V43.5/v43.5.ind\",\n",
    "          \"snpname\": \"/n/groups/reich/matt/pipeline/static/1240kSNP.snp\",\n",
    "          \"bamname\": \"FILL_IN\",\n",
    "          \"threshtable\": \"/n/groups/reich/matt/pipeline/static/pulldown_thresholds\",       \n",
    "          \"defstring\": \"capture_half\",\n",
    "          \"readbam\": \"/n/groups/reich/matt/pipeline/static/readbam\",\n",
    "          \"oldpullmode\": \"YES\",\n",
    "          \"sampname\": \"SAMP\",\n",
    "          \"printcount\": \"NO\",\n",
    "          \"majmode\": \"NO\",\n",
    "          \"rgonly\": \"NO\",\n",
    "          \"chrom\": \"24\",\n",
    "          \"readdepthname\": \"/n/groups/reich/hringbauer/git/y_chrom/output/SAMP.txt\"\n",
    "          }\n",
    "    return par_dc"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Run for a single Individual"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/n/data1/hms/genetics/reich/1000Genomes/amh_samples/ancientMergeSets__CAPTURE/C-per_sample_versions/I10266/v0030.2__2018_05_02/merged/by_chr/*Y.bam\n",
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/I10266\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/I10266\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: I10266\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: /n/groups/reich/matt/pipeline/static/1240kSNP.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ancientMergeSets__CAPTURE/C-per_sample_versions/I10266/v0030.2__2018_05_02/merged/by_chr/*Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: I10266\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/y_chrom/output/I10266.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 1909175941\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: I10266\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: /n/groups/reich/matt/pipeline/static/1240kSNP.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ancientMergeSets__CAPTURE/C-per_sample_versions/I10266/v0030.2__2018_05_02/merged/by_chr/*Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/y_chrom/output/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "number of valid snps: 32670\n",
      "sample I10266 is male\n",
      "numsnps: 1233013  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for I10266 (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0          I10266  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ancientMergeSets__CAPTURE/C-per_sample_versions/I10266/v0030.2__2018_05_02/merged/by_chr/*Y.bam Y \n",
      "triallelic: 22\n",
      "rawreads: 20805\n",
      "reg: 24  reads: 20805\n",
      "chrom 24 processed\n",
      "              I10266 Israel_MLBA_Canaanite mean depth:     0.120355 coverage: 3288\n",
      "histogram of read return codes\n",
      "read return code   -5 ::      3769\n",
      "read return code   -3 ::     13538\n",
      "read return code   -2 ::        23\n",
      "read return code   -1 ::        88\n",
      "read return code    1 ::      3387\n",
      "total: 20805\n",
      "\n",
      "## total reads: 20805  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 1233013\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 3288 invalids 0\n",
      "\n",
      "##end of pulldown:        7.970 seconds cpu      759.505 Mbytes in use\n",
      "CPU times: user 558 ms, sys: 119 ms, total: 678 ms\n",
      "Wall time: 8.67 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "#anno_path = \"/n/groups/reich/hringbauer/explore_ntbk/v43/V43/v42.3.anno\"\n",
    "anno_path = \"/n/groups/reich/DAVID/V43/V43.5/v43.5.anno\"\n",
    "df = pd.read_csv(anno_path, sep='\\t', low_memory=False)\n",
    "\n",
    "par_dc = give_default_par_dc()\n",
    "### Create a parfile to run\n",
    "savepath = create_parfile_from_ref(df=df, \n",
    "                                   iid=\"I10266\",  #I10266\n",
    "                                   par_dc=par_dc)\n",
    "### Produce Lots of Output (toggled)\n",
    "run_pulldown_y(path_parfile=savepath)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Do Multiple Individuals"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Doing SEC002\n",
      "/n/data1/hms/genetics/reich/1000Genomes/amh_samples/marcus2019_sardinia_samples/A-round1/B-fix/SEC002/hg19/SEC002.md_no1kgvar.bam.by_chr/*Y.bam\n",
      "Successfully saved to /n/groups/reich/hringbauer/git/y_chrom/parfiles/SEC002\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/marcus2019_sardinia_samples/A-round1/B-fix/SEC002/hg19/SEC002.md_no1kgvar.bam.by_chr/*Y.bam Y \n"
     ]
    }
   ],
   "source": [
    "#iids = [\"I7116\", \"I7121\", \"I7120\"]  # African R1b\n",
    "#iids = [\"R6.SG\", \"I18872\", \"I15941\",\"I8016\",\"I6752\", \"I16339\", \"I14337\", \"I8057\", \"I7116\"] #R1b-V88 canditates\n",
    "iids = [\"SEC002\"]\n",
    "\n",
    "\n",
    "for iid in iids:\n",
    "    print(f\"Doing {iid}\")\n",
    "    savepath = create_parfile_from_ref(df=df, iid=iid,\n",
    "                            par_dc=par_dc)\n",
    "    \n",
    "    run_pulldown_y(path_parfile=savepath, output=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Run the Punic Individuals"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#punic_df = pd.read_csv(\"../punic_aDNA/data/males_feb20.csv\")\n",
    "punic_df = pd.read_csv(\"../punic_aDNA/data/males_sic_feb20.csv\")\n",
    "\n",
    "male_punic_iids = punic_df.iloc[:,0].values\n",
    "\n",
    "for iid in male_punic_iids:\n",
    "    print(f\"Doing {iid}\")\n",
    "    savepath = create_parfile_from_ref(df=df, iid=iid,\n",
    "                            par_dc=par_dc)\n",
    "    \n",
    "    run_pulldown_y(path_parfile=savepath, output=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Run the Ning Individuals"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded 55 from /n/groups/reich/hringbauer/git/tibet_aDNA/Data/iids_male_ning2020.csv\n"
     ]
    }
   ],
   "source": [
    "iid_path = \"/n/groups/reich/hringbauer/git/tibet_aDNA/Data/iids_male_ning2020.csv\"\n",
    "iids = np.loadtxt(iid_path, dtype=\"str\")\n",
    "print(f\"Loaded {len(iids)} from {iid_path}\")\n",
    "\n",
    "anno_path = \"/n/groups/reich/DAVID/V43/V43.5/v43.5.anno\"\n",
    "df = pd.read_csv(anno_path, sep='\\t', low_memory=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/n/data1/hms/genetics/reich/1000Genomes/amh_samples/marcus2019_sardinia_samples/A-round1/B-fix/MA110/hg19/MA110.md_no1kgvar.bam.by_chr/*Y.bam\n",
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/MA110\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/MA110\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: MA110\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: /n/groups/reich/matt/pipeline/static/1240kSNP.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/marcus2019_sardinia_samples/A-round1/B-fix/MA110/hg19/MA110.md_no1kgvar.bam.by_chr/*Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: MA110\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: ./output/ning20/MA110.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 1438217005\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: MA110\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: /n/groups/reich/matt/pipeline/static/1240kSNP.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/marcus2019_sardinia_samples/A-round1/B-fix/MA110/hg19/MA110.md_no1kgvar.bam.by_chr/*Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: ./output/ning20/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "number of valid snps: 32670\n",
      "sample MA110 is male\n",
      "numsnps: 1233013  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for MA110 (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0           MA110  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/marcus2019_sardinia_samples/A-round1/B-fix/MA110/hg19/MA110.md_no1kgvar.bam.by_chr/*Y.bam Y \n",
      "triallelic: 453\n",
      "rawreads: 97600\n",
      "reg: 24  reads: 97600\n",
      "chrom 24 processed\n",
      "               MA110 Italy_Sardinia_BA_Nuragic_o mean depth:     2.326783 coverage: 21682\n",
      "histogram of read return codes\n",
      "read return code   -5 ::     27324\n",
      "read return code   -2 ::     11172\n",
      "read return code   -1 ::      1062\n",
      "read return code    1 ::     58042\n",
      "total: 97600\n",
      "\n",
      "## total reads: 97600  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 1233013\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 21682 invalids 0\n",
      "\n",
      "##end of pulldown:        8.680 seconds cpu      759.505 Mbytes in use\n",
      "CPU times: user 221 ms, sys: 60.4 ms, total: 282 ms\n",
      "Wall time: 8.95 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "par_dc = give_default_par_dc()\n",
    "### Create a parfile to run\n",
    "for iid in iids[:1]:\n",
    "    ### Produce the Y chromosome\n",
    "    savepath = create_parfile_from_ref(df=df, \n",
    "                                       iid=\"MA110\",  #I10266\n",
    "                                       par_dc=par_dc,\n",
    "                                       outfolder=\"./output/ning20/\")\n",
    "    run_pulldown_y(path_parfile=savepath) # Run the Pulldown"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Successfully saved parfile to /n/groups/reich/hringbauer/git/y_chrom/parfiles/ZLNR-2\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/ZLNR-2\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: ZLNR-2\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: /n/groups/reich/matt/pipeline/static/1240kSNP.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/B-fix/ZLNR-2/hg19/aln.mapped.rmdupse.md_no1kgvar.rg.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: ZLNR-2\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: ./output/ning20/ZLNR-2.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 1695689265\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: ZLNR-2\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/DAVID/V43/V43.5/v43.5.ind\n",
      "snpname: /n/groups/reich/matt/pipeline/static/1240kSNP.snp\n",
      "bamname: /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/B-fix/ZLNR-2/hg19/aln.mapped.rmdupse.md_no1kgvar.rg.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: ./output/ning20/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "number of valid snps: 32670\n",
      "numsnps: 1233013  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for ZLNR-2 (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0          ZLNR-2  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view /n/data1/hms/genetics/reich/1000Genomes/amh_samples/ning2020__eastAsian_samples/A-round1/B-fix/ZLNR-2/hg19/aln.mapped.rmdupse.md_no1kgvar.rg.bam Y \n",
      "triallelic: 0\n",
      "rawreads: 0\n",
      "*** no data... database problem? ***\n",
      "reg: 24  reads: 0\n",
      "chrom 24 processed\n",
      "              ZLNR-2          China_AR_EN mean depth:     0.000000 coverage: 0\n",
      "histogram of read return codes\n",
      "total: 0\n",
      "\n",
      "## total reads: 0  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 1233013\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 0 invalids 0\n",
      "\n",
      "##end of pulldown:        8.310 seconds cpu      759.505 Mbytes in use\n",
      "CPU times: user 205 ms, sys: 58.7 ms, total: 264 ms\n",
      "Wall time: 8.62 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "par_dc = give_default_par_dc()\n",
    "### Create a parfile to run\n",
    "for iid in iids[:1]:\n",
    "    ### Produce the Y chromosome\n",
    "    savepath = create_parfile_from_ref(df=df, \n",
    "                                       iid=iid,  #I10266\n",
    "                                       par_dc=par_dc,\n",
    "                                       outfolder=\"./output/ning20/\")\n",
    "    run_pulldown_y(path_parfile=savepath) # Run the Pulldown"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Run the Full ISOGG SNP set "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../*Y.bam\n",
      "Successfully saved to /n/groups/reich/hringbauer/git/y_chrom/parfiles/Loschbour_published.DG\n",
      "parameter file: /n/groups/reich/hringbauer/git/y_chrom/parfiles/Loschbour_published.DG\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: Loschbour_published.DG\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/hringbauer/explore_ntbk/v42/V42.3/v42.3.ind\n",
      "snpname: /n/groups/reich/hringbauer/git/y_chrom/data/eigenstrat/y_snps_all2020.snp\n",
      "bamname: ../*Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: Loschbour_published.DG\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/y_chrom/output/pulldowns_all/Loschbour_published.DG.txt\n",
      "version: 2400\n",
      "majmode unset\n",
      "old pullmode\n",
      "seed: 1354517211\n",
      "##start:paramfile\n",
      "BASE: /n/groups/reich/matt/pipeline/static\n",
      "SAMP: Loschbour_published.DG\n",
      "D2: ./output\n",
      "indivname: /n/groups/reich/hringbauer/explore_ntbk/v42/V42.3/v42.3.ind\n",
      "snpname: /n/groups/reich/hringbauer/git/y_chrom/data/eigenstrat/y_snps_all2020.snp\n",
      "bamname: ../*Y.bam\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds\n",
      "defstring: capture_half\n",
      "readbam: /n/groups/reich/matt/pipeline/static/readbam\n",
      "oldpullmode: YES\n",
      "sampname: SAMP\n",
      "printcount: NO\n",
      "majmode: NO\n",
      "rgonly: NO\n",
      "chrom: 24\n",
      "readdepthname: /n/groups/reich/hringbauer/git/y_chrom/output/pulldowns_all/SAMP.txt\n",
      "##end:paramfile\n",
      "##start:threshtable\n",
      "##             X      map      basequal        depth    skip  paired\n",
      "         ancient2  30  999  20  999   0  999  5  0\n",
      "         ancient  30  999  30  999   0  99  2  0\n",
      "         capture  10  999  20  999   0  99  2  0\n",
      "         capture_minus  10  999  20  999   0  99  10  0\n",
      "         capture_half  10  999  20  999   0  99  2  0\n",
      "         capture_plus  10  999  20  999   0  99  0  0\n",
      "##end:threshtable\n",
      "udgtype: -1\n",
      "bam_database: (null)\n",
      "/n/groups/reich/hringbauer/git/y_chrom/data/eigenstrat/y_snps_all2020.snp: genetic distance set from physical distance\n",
      "number of valid snps: 72933\n",
      "sample hed.DG is male\n",
      "numsnps: 72933  numindivs:  1\n",
      "reference: /n//groups/reich/reference-genomes/human_hg19/human_g1k_hs37d5/unzipped/hs37d5.fa\n",
      "polymask: /home/np29/broaddatax/tables/1Kpoly.fa\n",
      "default thresholds set for Loschbour_published.DG (using capture_half)\n",
      "threshtable: /n/groups/reich/matt/pipeline/static/pulldown_thresholds loaded\n",
      "poplist:  0 Loschbour_published.DG  10 999    20 999    0  99    2   0\n",
      "+++ reg 24 loaded;  length 59373566\n",
      "+++ polymask 24 loaded;  length 59373566\n",
      "/home/np29/o2bin/samtools view ../*Y.bam Y \n",
      "[E::hts_open_format] Failed to open file ../*Y.bam\n",
      "samtools view: failed to open \"../*Y.bam\" for reading: No such file or directory\n",
      "triallelic: 0\n",
      "rawreads: 0\n",
      "*** no data... database problem? ***\n",
      "reg: 24  reads: 0\n",
      "chrom 24 processed\n",
      "Loschbour_published.DG Luxembourg_Loschbour_published.DG mean depth:     0.000000 coverage: 0\n",
      "histogram of read return codes\n",
      "total: 0\n",
      "\n",
      "## total reads: 0  long reads: 0\n",
      "*** warning output snpname NULL\n",
      "snpname: (null) 72933\n",
      "indname:  (null) 1\n",
      "gname: (null)\n",
      "eigenstrat output\n",
      "valid genotypes 0 invalids 0\n",
      "\n",
      "##end of pulldown:        8.170 seconds cpu       46.801 Mbytes in use\n",
      "CPU times: user 187 ms, sys: 38.3 ms, total: 225 ms\n",
      "Wall time: 8.53 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "### Create a parfile to run\n",
    "savepath = create_parfile_from_ref(df=df, \n",
    "                                   iid=\"Loschbour_published.DG\",  #I10266\n",
    "                                   par_dc=par_dc,\n",
    "                                   \n",
    "                                   use_fullisogg=True)\n",
    "### Produce Lots of Output (toggled)\n",
    "run_pulldown_y(path_parfile=savepath)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Area 51"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Print old Parfile\n",
    "path = df1['Data: pulldown logfile location'].values[0]\n",
    "file = open(path,mode='r')\n",
    "txt = file.read()\n",
    "print(txt)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
